src/main.c

```diff
@@ -39,6 +39,41 @@ static inline void freep(void *p) {
 	free(*(void **)p);
 }
 
+int _already_in_systemd(pid_t pid, const char *target_slice) {
+	char path[PATH_MAX];
+	char buf[LINE_LEN];
+	char pattern[UNIT_NAME_LEN + 2];
+	FILE *f;
+	int r;
+
+	r = snprintf(path, PATH_MAX, "/proc/%i/cgroup", pid);
+	if (r < 0)
+		return r;
+
+	r = snprintf(pattern, UNIT_NAME_LEN + 2, "/%s/", target_slice);
+	if (r < 0)
+		return r;
+
+	f = fopen(path, "r");
+	if (!f)
+		return -errno;
+
+	while (fgets(buf, LINE_LEN, f)) {
+		if (strncmp(buf, "0:", 2) != 0)
+			continue;
+
+		if (strstr(buf, pattern)) {
+			r = 1;
+			goto final;
+		}
+	}
+	r = 0;
```

Comment:
nit: Move before the loop and replace `goto` with `break`?

Reply:
Changed in the new commit.